suite/volume/volume_setup.go

```diff
@@ -0,0 +1,320 @@
+type roundVals struct {
+	RoundID int64
+	Val     int64
+}
+
+// JobByInstance helper struct to match job + instance ID against prom metrics
+type JobByInstance struct {
+	ID       string
+	Instance string
+}
+
+// NewFluxTest deploys AggregatorsNum flux aggregators concurrently
+func NewFluxTest(spec *FluxTestSpec) (*FluxTest, error) {
+	s, err := suite.DefaultLocalSetup(spec.InitFunc)
+	if err != nil {
+		return nil, err
+	}
+	clNodes, nodeAddrs, err := suite.ConnectToTemplateNodes()
+	if err != nil {
+		return nil, err
+	}
+	err = suite.FundTemplateNodes(s.Client, s.Wallets, clNodes, 9e18, 0)
+	if err != nil {
+		return nil, err
+	}
+	s.Client.(*client.EthereumClient).BorrowedNonces(true)
+	adapter, err := tools.NewExternalAdapter()
+	if err != nil {
+		return nil, err
+	}
+
+	fluxInstances := make([]contracts.FluxAggregator, 0)
+	nodesByHostPort := make(map[string]client.Chainlink)
+	contractToJobsMap := make(map[string][]JobByInstance)
+	mu := &sync.Mutex{}
+	g := &errgroup.Group{}
+	for i := 0; i < spec.AggregatorsNum; i++ {
+		deployFluxInstance(&FluxInstanceDeployment{
+			InstanceDeployment: InstanceDeployment{
+				Index:   i,
+				Suite:   s,
+				Spec:    spec,
+				Oracles: nodeAddrs,
+				Nodes:   clNodes,
+				Adapter: adapter,
+			},
+			FluxInstancesMu:   mu,
+			NodesByHostPort:   nodesByHostPort,
+			FluxInstances:     &fluxInstances,
+			ContractToJobsMap: contractToJobsMap,
+		}, g)
+	}
+	err = g.Wait()
+	if err != nil {
+		return nil, err
+	}
+	prom, err := tools.NewPrometheusClient(s.Config.Prometheus)
+	if err != nil {
+		return nil, err
+	}
+	log.Debug().Interface("Contracts to jobs", contractToJobsMap).Msg("Debug data for per round metrics")
+	return &FluxTest{
+		Test: Test{
+			DefaultSetup:            s,
+			OnChainCheckAttemptsOpt: spec.OnChainCheckAttemptsOpt,
+			Nodes:                   clNodes,
+			Adapter:                 adapter,
+			Prom:                    prom,
+		},
+		FluxInstances:      &fluxInstances,
+		ContractsToJobsMap: contractToJobsMap,
+		NodesByHostPort:    nodesByHostPort,
+		roundsDurationData: make([]float64, 0),
+		AlreadySeenRuns:    make(map[string]bool),
+	}, nil
+}
+
+// deployFluxInstance deploy one flux instance concurrently, add jobs to all the nodes
+func deployFluxInstance(d *FluxInstanceDeployment, g *errgroup.Group) {
+	g.Go(func() error {
+		log.Info().Int("Instance ID", d.Index).Msg("Deploying contracts instance")
+		fluxInstance, err := d.Suite.Deployer.DeployFluxAggregatorContract(d.Suite.Wallets.Default(), d.Spec.FluxOptions)
+		if err != nil {
+			return err
+		}
+		err = fluxInstance.Fund(d.Suite.Wallets.Default(), big.NewInt(0), big.NewInt(1e18))
+		if err != nil {
+			return err
+		}
+		err = fluxInstance.UpdateAvailableFunds(context.Background(), d.Suite.Wallets.Default())
+		if err != nil {
+			return err
+		}
+		// set oracles and submissions
+		err = fluxInstance.SetOracles(d.Suite.Wallets.Default(),
+			contracts.SetOraclesOptions{
```

Comment:
Name of this struct implies that it's generic to all contract types. Should be named `FluxAggregatorOracleOptions`.